intent/src/lib.rs

```diff
@@ -103,16 +121,47 @@ impl FungibleTokenReceiver for IntentContractImpl {
         // self.assert_token(&sender_id); // TODO: Check if we need tokens validation.
         let action = Action::decode(msg).expect("decode Action");
 
-        match action {
+        let promise = match action {
             Action::CreateIntent(id, intent) => {
                 log!(
                     "Creating the intent with id: {id} by: {sender_id}, amount: {}",
                     amount.0
                 );
-                self.create_intent(id, amount, intent).unwrap()
+
+                require!(id.len() <= 128, "ID is too long");
+
+                // First check that initiator has storage deposit on token he wants to get.
+                ext_storage_management::ext(intent.receive.token_id.clone())
+                    .storage_balance_of(sender_id)
+                    .then(
+                        Self::ext(env::current_account_id())
+                            .with_static_gas(FINISH_CREATING_GAS)
+                            .finish_creating_intent(id, amount, intent),
```

Comment:
maybe check in advance that we don't have an intent with such id? or create it with locked status?

Reply:
But we do [it](https://github.com/defuse-is-near/defuse-contracts/blob/fix/storage_deposit/intent/src/lib.rs#L332) in the `create_intent` method.

Reply:
`create_intent` is called only in `finish_creating_intent`. By making this check only after response from `storage_balance_of` we just spend user's gas while we can fail early